pkg/plugins/apiserver_test.go

```diff
@@ -7,15 +7,16 @@ import (
 )
 
 func TestUtils(t *testing.T) {
-	// multiple flavors of the same idea
-	require.Equal(t, "tempo.datasource.grafana.app", getIDIgnoreError("tempo"))
-	require.Equal(t, "tempo.datasource.grafana.app", getIDIgnoreError("grafana-tempo-datasource"))
-	require.Equal(t, "tempo.datasource.grafana.app", getIDIgnoreError("tempo-datasource"))
+	// all other datasources use plugin ID as prefix (as long as is valid)
+	require.Equal(t, "name.datasource.grafana.app", getIDIgnoreError("name"), "single word is accepted")
+	require.Equal(t, "name-datasource.datasource.grafana.app", getIDIgnoreError("name-datasource"), "multiple words are accepted with '-datasource' suffix")
+	require.Error(t, getErrorIgnoreValue("org-name"), "multiple words are not accepted without '-datasource' suffix")
```

Comment:
I don't know if it matters but we could have `azure-monitor`, which is just the name but it includes a `-`.